File: jpa-separate-ro-rw/src/test/java/com/sight/testType/DataJpaRepoTest.java
package com.sight.testType;

import com.sight.config.db.board.BoardDataSourceConfig;
import com.sight.config.db.board.BoardDatabaseConfig;
import com.sight.testConfig.JpaTestConfig;
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.context.annotation.Import;

import javax.transaction.Transactional;

@DataJpaTest
@Transactional
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
@Import(value = {
    JpaTestConfig.class,
    BoardDatabaseConfig.class,
    BoardDataSourceConfig.class
})
public abstract class DataJpaRepoTest implements BaseTest {

}

File: JPAwithRecursive/src/test/kotlin/com/sight/blog/config/test/DataJpaRepoTest.kt
package com.sight.blog.config.test

import com.sight.blog.common.jpa.criteria.CriteriaConfig
import com.sight.blog.config.TestEnv
import com.sight.blog.config.jpa.JpaTestConfig
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
import org.springframework.context.annotation.Import
import org.springframework.test.context.ActiveProfiles
import org.springframework.transaction.annotation.Transactional

@DataJpaTest
@Transactional
@ActiveProfiles(TestEnv.TEST)
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
@Import(value = [JpaTestConfig::class, CriteriaConfig::class])
abstract class DataJpaRepoTest: BaseTest